### DISCUSSION TITLE
Error when trying to introduce new services after OpenELB installation

### DISCUSSION THREAD

**Discussion by hustluh** [openelb/openelb#382] [General]
I just installed OpenELB as our on-prem load balancer for a single-node, single-cluster environment.

After verifying our ingress-nginx controller service received an IP, I tried to add a service for our front end only to receive the following error:

```
Error from server (InternalError): error when creating "test-frontend.yaml": Internal error occurred: failed calling webhook "mutating.eip.network.kubesphere.io": failed to call webhook: Post "https://openelb-admission.openelb-system.svc:443/validate-network-kubesphere-io-v1alpha2-svc?timeout=10s": tls: failed to verify certificate: x509: certificate is valid for ingress.local, not openelb-admission.openelb-system.svc
```

Any idea how to fix this?

**Comment by renyunkang** [comment]
How did you installed openelb?

Can you provide the contents of test-frontend.yaml?

**Comment by hustluh** [comment]
By following this guide::
https://openelb.io/docs/getting-started/installation/install-openelb-on-kubernetes/

**Comment by renyunkang** [comment]
We use a job to generate a secret and store it in the secret resource openelb-system/openelb-admission, which is then mounted into openelb-manager. It seems like there might be an issue with the secret.

**Comment by renyunkang** [comment]
Could you provide the secret openelb-system/openelb-admission?
`kubectl get secret -n openelb-system openelb-admission -o yaml`

**Comment by hustluh** [comment]
How can I troubleshoot it?

**Comment by hustluh** [comment]
I will do this first thing in the morning

**Comment by hustluh** [comment]
@renyunkang here is the output

```
[root@mrte1 ~]# kubectl get secret -n openelb-system openelb-admission -o yaml
apiVersion: v1
data:
  ca: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJkVENDQVJ5Z0F3SUJBZ0lSQUpYR2pnYlFVbE44U2pOSm1HVkxCZWt3Q2dZSUtvWkl6ajBFQXdJd0R6RU4KTUFzR0ExVUVDaE1FYm1sc01UQWdGdzB5TkRBeE1qVXhPRFV4TURCYUdBOHlNVEkwTURFd01URTROVEV3TUZvdwpEekVOTUFzR0ExVUVDaE1FYm1sc01UQlpNQk1HQnlxR1NNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJIazlMdW9sCk9TbTIvTGxoVS9SUzNvUEVvZUx3RWJobWpXSDRUVTEvbHRvVHBHaGJvMHRrSUhxNFlMUGJ5dWJCUThGeEdRWUsKRTBWRXhuQ2RQYTJoaWhhalZ6QlZNQTRHQTFVZER3RUIvd1FFQXdJQ0JEQVRCZ05WSFNVRUREQUtCZ2dyQmdFRgpCUWNEQVRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUIwR0ExVWREZ1FXQkJRNWFpdDFoaFhDc1Zuc1Q1YjhFTENlCkZXeWlOakFLQmdncWhrak9QUVFEQWdOSEFEQkVBaUFGV3VyK0tsRklpQ1cwd3cwUWk2YVRkbGZDNlVudUV2aS8KNnpGMm5IUTkyd0lnTmc5RGE1WFRXMHpqU1pRVzdjZjJwZjRXOVJ6akg1VkJrVnlOOExmUmtpYz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
  cert: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJtRENDQVQyZ0F3SUJBZ0lRQVFKeVdLRUlpKzRoSUFvMStRdktKVEFLQmdncWhrak9QUVFEQWpBUE1RMHcKQ3dZRFZRUUtFd1J1YVd3eE1DQVhEVEkwTURFeU5URTROVEV3TUZvWUR6SXhNalF3TVRBeE1UZzFNVEF3V2pBUApNUTB3Q3dZRFZRUUtFd1J1YVd3eU1Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMERBUWNEUWdBRTNQcFpBL2txCmxaTSt2WmRoV1YwQnJBSmVJV2Z4L3UvejBnZDdLQ0E2YzlhSGRCTy9Ob3lGZTFhNGRGa0RVeWovZk9kNWZIZDQKYmtaU1ZYdkJRUlZNSUtONU1IY3dEZ1lEVlIwUEFRSC9CQVFEQWdXZ01CTUdBMVVkSlFRTU1Bb0dDQ3NHQVFVRgpCd01CTUF3R0ExVWRFd0VCL3dRQ01BQXdRZ1lEVlIwUkJEc3dPWUlSYjNCbGJtVnNZaTFoWkcxcGMzTnBiMjZDCkpHOXdaVzVsYkdJdFlXUnRhWE56YVc5dUxtOXdaVzVsYkdJdGMzbHpkR1Z0TG5OMll6QUtCZ2dxaGtqT1BRUUQKQWdOSkFEQkdBaUVBM2RVSGhTb2xmQnVrcDFWYnE4VDZXUUo0K2RxTWt1WkhCT2ZKdnd6OHJ6UUNJUUNBN2tkMApadk5ESGF3blBkbHN3M0t4bVpZeGdoTzBwYzFmV2U5S0IrM0xTdz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
  key: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1IY0NBUUVFSUJUeHpPSHdNbHQvOWwwRzJ6UTFJU1E4R1VLTitDc0RpMjdyMVNsbUQyOHBvQW9HQ0NxR1NNNDkKQXdFSG9VUURRZ0FFM1BwWkEva3FsWk0rdlpkaFdWMEJyQUplSVdmeC91L3owZ2Q3S0NBNmM5YUhkQk8vTm95RgplMWE0ZEZrRFV5ai9mT2Q1ZkhkNGJrWlNWWHZCUVJWTUlBPT0KLS0tLS1FTkQgRUMgUFJJVkFURSBLRVktLS0tLQo=
kind: Secret
metadata:
  creationTimestamp: "2024-01-25T18:56:00Z"
  name: openelb-admission
  namespace: openelb-system
  resourceVersion: "29515"
  uid: ee9ded0c-143a-46a0-963c-82c39894a5ca
type: Opaque
```

**Reply by renyunkang** [reply]
The generated OpenELB certificate is valid. Refer to [this](https://gitlab.com/gitlab-org/charts/gitlab/-/issues/5124) and continue investigating whether the issue might be related to the Ingress configuration.

![image](https://github.com/openelb/openelb/assets/33660223/fd61cf7f-6f83-4bbc-9c20-4127f99ee615)
![image](https://github.com/openelb/openelb/assets/33660223/858998b6-a020-4154-afbf-c21e875aa897)

**Comment by hustluh** [comment]
would I add the secret for openelb-admission to ingress-nginx-admission?

As a side note, I am attempting to pass the client's real IP to our API service behind the ingress controller. I was instructed by the k8s community to install an LB (I selected openELB vs metalLB) to assign an IP to the ingress-nginx-controller, which would then allow everything else to function as expected. Is openELB trying to act as the ingress controller as well?
